### DISCUSSION TITLE
No GPS navigation

### DISCUSSION THREAD

**Discussion by yuval-herman** [osmandapp/OsmAnd#18889] [Ideas]
I was navigating in some areas without any reception at all (including no GPS reception) when some idea popped into my head.
There are a couple if methods for navigating with a GPS fix but the usual issue with them is the over time drift acquired.
My idea was using the known road structure to correct drift over time.
Basically, we use [dead reckoning](https://en.m.wikipedia.org/wiki/Dead_reckoning) methods normally but also use roads to correct for drift.
When we go over a noticable turn and our calculation is being slightly off we can correct it.

**Comment by Sonwon1** [comment]
For OsmAnd to know it missed a corner requires a gps fix so there is that. It could assume road speed continued within limits and plot your location over time. I think my Garmin did better in tunnels than OsmAnd.
